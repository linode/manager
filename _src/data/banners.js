import { BANNER_TYPES } from '~/constants';

const {
  OUTAGE,
  REBOOT_SCHEDULED,
  BALANCE_OUTSTANDING,
  TICKET_IMPORTANT,
  TICKET_ABUSE,
} = BANNER_TYPES;

export const testApiBanner = {
  entity: {
    id: 1234,
    type: 'ticket',
    label: 'testing ticket',
  },
  type: TICKET_IMPORTANT,
  when: null,
};

export const testBillingBanner = {
  ...testApiBanner,
  entity: {
    id: 1235,
    label: 'Please update your billing information.',
  },
};

export const outstandingBalanceBanner = {
  entity: {
    id: null,
    label: 'user@email.com',
    type: 'account',
    url: '/account/settings',
  },
  type: BALANCE_OUTSTANDING,
  when: null,
};

export const scheduledRebootBanner = {
  entity: {
    id: 1234,
    label: 'my-linode',
    type: 'linode',
    url: '/linode/instances/1234',
  },
  type: REBOOT_SCHEDULED,
  when: '2017-12-31T23:59:59',
};

export const testAbuseBanner = {
  ...testApiBanner,
  entity: {
    id: 1235,
    label: 'ToS Violation',
  },
  type: TICKET_ABUSE,
};

export const testAbuseBanner2 = {
  ...testAbuseBanner,
  entity: {
    id: 1236,
  },
};

const testOutageBanner1 = {
  when: null,
  entity: {
    label: null,
    id: 'us-east',
    type: 'region',
    url: '/regions/us-east',
  },
  type: OUTAGE,
};

const testOutageBanner2 = {
  ...testOutageBanner1,
  entity: {
    ...testOutageBanner1.entity,
    id: 'us-central',
    url: '/regions/us-central',
  },
};

export const testGlobalBanner = {
  entity: null,
  message: 'The new linode cloud manager is in the works!',
  type: 'linode',
  when: '2017-12-22T05:00:00',
};

export const importantBanner = [
  testApiBanner,
];

export const importantBanners = [
  testApiBanner,
  testBillingBanner,
];

export const abuseBanner = [
  testAbuseBanner,
];

export const abuseBanners = [
  testAbuseBanner,
  testAbuseBanner2,
];

export const outageBanners = [
  testOutageBanner1,
  testOutageBanner2,
];

export const globalBanner = [
  testGlobalBanner,
];

export const banners = [
  testApiBanner,
  testBillingBanner,
  testAbuseBanner,
  testGlobalBanner,
];
