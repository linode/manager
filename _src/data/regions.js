export const apiTestRegion = {
  id: 'us-east',
  country: 'US',
  label: 'Newark, NJ',
};

export const testRegion = {
  ...apiTestRegion,
  _polling: false,
};

export const regions = {
  'us-east': testRegion,
  // TODO: The alpha env only has Newark, but maybe we want to add more DCs
  // here later anyway
};
