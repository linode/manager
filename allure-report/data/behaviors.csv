"Epic","Feature","Story","FAILED","BROKEN","PASSED","SKIPPED","UNKNOWN"
"","","","2","0","10","0","0"
