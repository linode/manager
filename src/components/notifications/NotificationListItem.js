import React, { PropTypes } from 'react';

import { Link } from 'react-router';
import TimeDisplay from '~/components/TimeDisplay';

import EventTypeMap from './EventTypes';


export default function NotificationListItem(props) {
  const event = props.event;
  const timeRemaining = event.time_remaining;
  const eventOptions = EventTypeMap[event.action];

  const baseCls = 'NotificationList-listItem';
  let className = baseCls;

  // This should not happen (it is a bug in the db). But if it does, don't blow up.
  if (!event.entity) {
    return null;
  }

  let entity;
  let message;
  if (!event.read) {
    entity = (<strong>{event.entity.label}</strong>);
    className += ` ${baseCls}--unread`;
  } else {
    entity = event.entity.label;
    className += ` ${baseCls}--read`;
  }

  if (timeRemaining !== null && timeRemaining > 0) {
    message = (
      <span>
        <span>
          {`${eventOptions.presentTenseAction} `}
          {entity}
        </span>
        <small className="text-muted">
          {/* TODO: Time remaining estimation */}
          {/* {` ${_rate} Estimated ${_time_remaining}`} */}
        </small>
      </span>
    );
  } else {
    const timestampMessage = (
      <div>
        <small>
          <span className="text-muted">
            <TimeDisplay time={event.created} /> by&nbsp;
          </span>
          <strong>{event.username}</strong>
        </small>
      </div>
    );

    // TODO: check against upcoming API Change
    if (event.status === 'failed') {
      message = (
        <span>
          <span>
            {`${eventOptions.presentTenseAction} `}
            {entity} failed
          </span>
          {timestampMessage}
        </span>
      );
    } else if (eventOptions.pastTensePrefix) {
      message = (
        <div>
          <div>
            {`${eventOptions.pastTensePrefix} `}
            {entity}
            {` ${eventOptions.pastTenseAction} `}
          </div>
          {timestampMessage}
        </div>
      );
    } else {
      message = (
        <span>
          <span>
            {entity}
            {` ${eventOptions.pastTenseAction} `}
          </span>
          {timestampMessage}
        </span>
      );
    }
  }

  return (
    <Link
      to={eventOptions.redirectUrl(event.entity)}
      className={className}
      onClick={() => props.onClick(event)}
    >
      {message}
    </Link>
  );
}

NotificationListItem.propTypes = {
  onClick: PropTypes.func,
};
