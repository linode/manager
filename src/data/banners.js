export const testApiBanner = {
  entity: {
    id: 1234,
    type: 'ticket',
    label: 'testing ticket',
  },
  type: 'important_ticket',
  when: null,
};

export const testBillingBanner = {
  ...testApiBanner,
  entity: {
    id: 1235,
    label: 'Please update your billing information.',
  },
};

export const testAbuseBanner = {
  ...testApiBanner,
  entity: {
    id: 1235,
    label: 'ToS Violation',
  },
  type: 'abuse_ticket',
};

export const testAbuseBanner2 = {
  ...testAbuseBanner,
  entity: {
    id: 1236,
  },
};

export const importantBanner = [
  testApiBanner,
];

export const importantBanners = [
  testApiBanner,
  testBillingBanner,
];

export const abuseBanner = [
  testAbuseBanner,
];

export const abuseBanners = [
  testAbuseBanner,
  testAbuseBanner2,
];

export const banners = [
  testApiBanner,
  testBillingBanner,
  testAbuseBanner,
];
