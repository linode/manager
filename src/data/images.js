export const apiTestImage = {
  id: 'linode/arch2016.05',
  recommended: true,
  vendor: 'Arch',
  label: 'Arch Linux 2016.05',
  size: 550,
  created: '2009-01-01T00:00:00',
  description: 'Test Image',
  type: 'manual',
  created_by: 'linode',
  is_public: true,
  deprecated: false,
};

export const testLinodeImage = {
  ...apiTestImage,
  _polling: false,
};

export const publicImages = {
  'linode/arch2016.05': testLinodeImage,
  'linode/debian7': {
    ...testLinodeImage,
    id: 'linode/debian7',
    vendor: 'Debian',
    label: 'Debian 7',
  },
  'linode/debian8.1': {
    ...testLinodeImage,
    id: 'linode/debian8.1',
    vendor: 'Debian',
    label: 'Debian 8.1',
  },
  'linode/debian6': {
    ...testLinodeImage,
    id: 'linode/debian6',
    recommended: false,
    vendor: 'Debian',
    label: 'Debian 6',
  },
};

export const testPrivateImage = {
  ...testLinodeImage,
  id: 'private/38',
  label: 'testImage',
  description: 'image details',
  vendor: null,
  created: '2017-08-08T13:55:16',
  created_by: 'zues',
  size: 2048,
  type: 'manual',
  is_public: false,
};

export const privateImages = {
  'private/38': testPrivateImage,
  'private/39': {
    ...testPrivateImage,
    id: 'private/39',
    label: 'nextImage',
    description: 'image from deleted disk',
    created: '2017-08-08T13:55:16',
    created_by: 'zues',
    size: 2048,
    type: 'automatic',
    is_public: false,
  },
};

export const images = {
  ...privateImages,
  ...publicImages,
};
