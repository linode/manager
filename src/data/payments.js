export const apiTestPayment = {
  usd: 5.0,
  id: 1000,
  date: '2015-11-12T16:11:59',
};

export const payments = [0, 1, 2, 3, 4].map(i => ({
  ...apiTestPayment,
  id: apiTestPayment.id + i,
  usd: apiTestPayment.usd + i,
}));
