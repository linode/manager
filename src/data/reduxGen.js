import { ONE, MANY, DELETE } from '~/api/internal.ts';

import { configsNodeBalancer, noGroupNodeBalancer } from '~/data/nodebalancers';

export const resource = { ...configsNodeBalancer };
export const page = {
  data: [configsNodeBalancer, noGroupNodeBalancer],
  pages: 1,
  results: 1,
  page: 1,
  nodebalancers: [configsNodeBalancer, noGroupNodeBalancer],
};

export const testConfigOne = {
  name: 'nodebalancers',
  primaryKey: 'id',
  endpoint: id => `/nodebalancers/${id}`,
  supports: [ONE],
};

export const testConfigMany = {
  name: 'nodebalancers',
  primaryKey: 'id',
  endpoint: id => `/nodebalancers/${id}`,
  supports: [MANY],
};

export const testConfigDelete = {
  name: 'nodebalancers',
  primaryKey: 'id',
  endpoint: id => `/nodebalancers/${id}`,
  supports: [DELETE],
};
