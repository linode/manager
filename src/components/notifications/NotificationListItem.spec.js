import React from 'react';
import { shallow } from 'enzyme';
import { StaticRouter } from 'react-router-dom';
import NotificationListItem from './NotificationListItem';

import { api } from '~/data';

const events = api.events;

describe('components/notifications/NotificationListItem', () => {
  function makeNotificationListItem(event = events.events[386]) {
    return (
      <StaticRouter>
        <NotificationListItem event={event} />
      </StaticRouter>
    );
  }

  it('renders unread notification', () => {
    const wrapper = shallow(makeNotificationListItem(events.events[385]));
    const notification = wrapper.dive().dive();

    expect(notification.find('.NotificationList-listItem--unread').length).toBe(1);
  });

  it('renders read notification', () => {
    const wrapper = shallow(makeNotificationListItem());
    const notification = wrapper.dive();

    expect(notification.find('.NotificationList-listItem--unread').length).toBe(0);
  });
});
