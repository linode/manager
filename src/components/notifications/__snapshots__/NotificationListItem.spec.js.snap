// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/notifications/NotificationListItem should render without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationListItem
    event={
      Object {
        "action": "linode_reboot",
        "created": "2016-11-21T20:03:40",
        "entity": Object {
          "label": "linode-www2",
        },
        "finished": null,
        "id": 386,
        "label_message": "",
        "linode_id": 1237,
        "nodebalancer_id": null,
        "percent_complete": 100,
        "read": true,
        "seen": false,
        "stackscript_id": null,
        "status": "finished",
        "updated": "2016-11-21T20:12:40",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <div>
          <strong>
            linode-www2
          </strong>
           rebooted 
        </div>
        <div>
          <small>
            <span
              className="text-muted"
            >
              <TimeDisplay
                time="2016-11-21T20:03:40"
              />
               by 
            </span>
            <strong />
          </small>
        </div>
      </div>,
      "className": "NotificationList-listItem NotificationList-listItem--read",
      "onClick": [Function],
      "replace": false,
      "to": "/linodes/linode-www2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <strong>
              linode-www2
            </strong>
             rebooted 
          </div>,
          <div>
            <small>
              <span
                className="text-muted"
              >
                <TimeDisplay
                  time="2016-11-21T20:03:40"
                />
                 by 
              </span>
              <strong />
            </small>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                linode-www2
              </strong>,
              " rebooted ",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "linode-www2",
              },
              "ref": null,
              "rendered": "linode-www2",
              "type": "strong",
            },
            " rebooted ",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <small>
              <span
                className="text-muted"
              >
                <TimeDisplay
                  time="2016-11-21T20:03:40"
                />
                 by 
              </span>
              <strong />
            </small>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="text-muted"
                >
                  <TimeDisplay
                    time="2016-11-21T20:03:40"
                  />
                   by 
                </span>,
                <strong />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <TimeDisplay
                      time="2016-11-21T20:03:40"
                    />,
                    " by ",
                  ],
                  "className": "text-muted",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "time": "2016-11-21T20:03:40",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " by ",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "strong",
              },
            ],
            "type": "small",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <div>
            <strong>
              linode-www2
            </strong>
             rebooted 
          </div>
          <div>
            <small>
              <span
                className="text-muted"
              >
                <TimeDisplay
                  time="2016-11-21T20:03:40"
                />
                 by 
              </span>
              <strong />
            </small>
          </div>
        </div>,
        "className": "NotificationList-listItem NotificationList-listItem--read",
        "onClick": [Function],
        "replace": false,
        "to": "/linodes/linode-www2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <strong>
                linode-www2
              </strong>
               rebooted 
            </div>,
            <div>
              <small>
                <span
                  className="text-muted"
                >
                  <TimeDisplay
                    time="2016-11-21T20:03:40"
                  />
                   by 
                </span>
                <strong />
              </small>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  linode-www2
                </strong>,
                " rebooted ",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "linode-www2",
                },
                "ref": null,
                "rendered": "linode-www2",
                "type": "strong",
              },
              " rebooted ",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <small>
                <span
                  className="text-muted"
                >
                  <TimeDisplay
                    time="2016-11-21T20:03:40"
                  />
                   by 
                </span>
                <strong />
              </small>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="text-muted"
                  >
                    <TimeDisplay
                      time="2016-11-21T20:03:40"
                    />
                     by 
                  </span>,
                  <strong />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <TimeDisplay
                        time="2016-11-21T20:03:40"
                      />,
                      " by ",
                    ],
                    "className": "text-muted",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "time": "2016-11-21T20:03:40",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " by ",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "strong",
                },
              ],
              "type": "small",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
